cppapi/client/event.cpp

```diff
@@ -1823,6 +1832,42 @@ int EventConsumer::connect_event(DeviceProxy *device,
 	return ret_event_id;
 }
 
+string EventConsumer::get_client_attribute_name(const string &local_callback_key)
+{
+	auto pos = local_callback_key.rfind('.');//remove event_type e.g. .idl5_change
+	return local_callback_key.substr(0, pos);
+}
+
+void Tango::EventConsumer::initialize_received_from_admin(const Tango::DevVarLongStringArray *dvlsa,
+														  const string &local_callback_key,
+														  const string &adm_name)
+{
+	auto tango_lib_ver = dvlsa->lvalue[0];
+
+	if (tango_lib_ver >= 936)
+	{
+		received_from_admin.event_name = (dvlsa->svalue[dvlsa->svalue.length() - 2]);
+	}
+	else
+	{
+		received_from_admin.event_name = std::move(local_callback_key);
+	}
+
+	if (tango_lib_ver >= 936)
+	{
+		received_from_admin.channel_name = (dvlsa->svalue[dvlsa->svalue.length() - 1]);
+	}
+	else
+	{
+		received_from_admin.channel_name = std::move(adm_name);
+	}
+
+	assert(not(received_from_admin.event_name.empty()));
```

Comment:
We should not follow Codacy recommendations here (or maybe we should find a way to tune them so it does not advise us to use this keyword).
"not"  keyword is not recognized by Visual C++ compilers.
This is why the appveyor build is broken.